can-auth/vatican.c

```diff
@@ -21,14 +21,41 @@
  * along with VulCAN. If not, see <http://www.gnu.org/licenses/>.
  */
 #include "vatican.h"
+#include "../drivers/irq_can.h"
+#include "../drivers/compare.c"
 #include <sancus_support/sm_io.h>
+#include <sancus_support/timer.h>
+#include <sancus_support/tsc.h>
+
+#define VATITACAN_NONCE_SIZE    2
+#define VATITACAN_DELTA	        1000
+#define VATITACAN_OFFSET	0
 
 // ============ VATICAN HELPER FUNCTIONS ============
 
 VULCAN_DATA ican_link_info_t  *vatican_connections;
 VULCAN_DATA size_t             vatican_nb_connections;
 VULCAN_DATA ican_link_info_t  *vatican_cur;
 
+DECLARE_TSC_TIMER(iat_timer);
```

Comment:
be aware that these timers are currently not necessarily trusted as the timer peripheral is also inaccessible by unprotected code right?